pandas/core/ops.py

```diff
@@ -582,6 +588,150 @@ def _get_op_name(op, special):
 DataFrame.{reverse}
 """
 
+_flex_comp_doc_FRAME = """
+Flexible wrappers to comparison operators (`eq`, `ne`, `le`, `lt`, `ge`, `gt`).
+
+Equivalent to `==`, `=!`, `<=`, `<`, `>=`, `>` with support to choose
+axis (rows or columns) and level for comparison.
+
+Parameters
+----------
+other : constant, list, tuple, ndarray, Series, or DataFrame
+    Any single or multiple element data structure, or list-like object.
+axis : int or str, optional
+    Axis to target. Can be either the axis name ('index', 'rows',
+    'columns') or number (0, 1).
+level : int or name
+    Broadcast across a level, matching Index values on the passed
+    MultiIndex level.
+
+Returns
+-------
+result : DataFrame of bool
+    Result of the comparison.
+
+See also
+--------
+DataFrame.eq : Compare DataFrames for equality elementwise
+DataFrame.ne : Compare DataFrames for inequality elementwise
+DataFrame.le : Compare DataFrames for less than inequality
+    or equality elementwise
+DataFrame.lt : Compare DataFrames for strictly less than
+    inequality elementwise
+DataFrame.ge : Compare DataFrames for greater than inequality
+    or equality elementwise
+DataFrame.gt : Compare DataFrames for strictly greater than
+    inequality elementwise
+
+Notes
+--------
+Mismatched indices will be unioned together.
+
+Examples
+--------
+>>> df1 = pd.DataFrame({'company': ['A', 'B', 'C'],
+...                     'cost': [250, 150, 100],
+...                     'revenue': [100, 250, 300]})
+>>> df1
+  company  cost  revenue
+0       A   250      100
+1       B   150      250
+2       C   100      300
+>>> df2 = pd.DataFrame({'company': ['A', 'B', 'C', 'D'],
+...                     'revenue': [300, 250, 100, 150]})
+>>> df2
+  company  revenue
+0       A      300
+1       B      250
+2       C      100
+3       D      150
```

Comment:
Instead if defining all the DataFrames first, I'd define the first, show the examples that don't need the others, and define the others immediately before they are used.

For the naming, I'd name `df` the first, `other` the second used to be compared. And something like `df_multiindex` the last one. I think that will make things easier for the users.

I also think that would make things easier for the user having smaller DataFrames (like 2x2). These big `True`/`False` DataFrame results are not immediate to understand.

Reply:
Understood on names. As for lengths, df1 and df2 have different number of rows and columns to show results of across different lengths. And it is just one more row and one more column. Surely not that *big*. I can add a note to mention results of different lengths.